test/extended/images/hardprune.go

```diff
@@ -475,22 +475,84 @@ func ConfigureRegistry(oc *exutil.CLI, desiredState RegistryConfiguration) error
 
 	err = oc.Run("env").Args(append([]string{"dc/docker-registry"}, envOverrides...)...).Execute()
 	if err != nil {
-		return fmt.Errorf("failed to update registry's environment with %s: %v", &waitForVersion, err)
+		return false, fmt.Errorf("failed to update registry's environment with %s: %v", &waitForVersion, err)
 	}
-	return exutil.WaitForDeploymentConfig(
+	err = exutil.WaitForDeploymentConfig(
 		oc.AdminKubeClient(),
 		oc.AdminAppsClient().Apps(),
 		metav1.NamespaceDefault,
 		"docker-registry",
 		waitForVersion,
 		oc)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func RedeployRegistry(oc *exutil.CLI) (bool, error) {
+	defer func(ns string) { oc.SetNamespace(ns) }(oc.Namespace())
+
+	oc = oc.SetNamespace(metav1.NamespaceDefault).AsAdmin()
+
+	env, err := oc.Run("env").Args("dc/docker-registry", "--list").Output()
+	if err != nil {
+		return false, err
+	}
+
+	num := 0
+
+	for _, envVar := range strings.Split(env, "\n") {
+		if len(envVar) == 0 {
+			continue
+		}
+
+		envKV := strings.SplitN(envVar, "=", 2)
+		if len(envKV) == 0 || envKV[0] != "REDEPLOY_REGISTRY" {
+			continue
+		}
+
+		i, err := strconv.Atoi(envKV[1])
+		if err != nil {
+			continue
```

Comment:
not sure why you continue here.  you won't see another env key "REDEPLOY_REGISTRY" in the loop.  might as well break.